### PR TITLE
[Backport 2.28] Fix incorrect dispatch to USAGE in example programs, which causes uninitialized memory to be used

### PR DESCRIPTION
**Pull request by aditya-deshpande-arm**
## Description

Fixes issues #1120 and #6700.

Numerous example programs contain the following:
```
if (argc == 0) {
    goto USAGE;
}

// later in program
p = argv[1];
```
If a program is called from the shell then`argc = 1`, so the program will not go to USAGE due to the check. The access to `argv[1]` is accessing undefined memory. This PR changes the check to `if (argc < 2)`.

## Gatekeeper checklist

- [x] **changelog** provided
- [x] **backport** of #6998 
- [x] **tests** not required, example programs

### CODE REVIEW

**Review by daverodgman — DISMISSED** [review]
LGTM

File: programs/test/query_compile_time_config.c [hunk 1]
```diff
@@ -38,7 +38,9 @@
 
 int main(int argc, char *argv[])
 {
-    if (argc != 2) {
+    int i;
```
Comment (tom-cosgrove-arm) [inline programs/test/query_compile_time_config.c:41]: where has this `int i` come from?

**Review by tom-cosgrove-arm — APPROVED** [review]
LGTM

File: programs/test/query_compile_time_config.c (hunk 1 above)
Comment (tom-cosgrove-arm) [reply programs/test/query_compile_time_config.c]: (now removed)

**Review by daverodgman — APPROVED** [review]

**Pull request merged**